core/src/main/java/com/sequenceiq/cloudbreak/job/salt/StackSaltStatusCheckerJob.java

```diff
@@ -0,0 +1,157 @@
+import com.sequenceiq.cloudbreak.reactor.api.event.cluster.RotateSaltPasswordReason;
+import com.sequenceiq.cloudbreak.service.RotateSaltPasswordService;
+import com.sequenceiq.cloudbreak.service.stack.StackDtoService;
+import com.sequenceiq.flow.core.FlowLogService;
+
+import io.opentracing.Tracer;
+
+@DisallowConcurrentExecution
+@Component
+public class StackSaltStatusCheckerJob extends StatusCheckerJob {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(StackSaltStatusCheckerJob.class);
+
+    @Inject
+    private StackSaltStatusCheckerJobService jobService;
+
+    @Inject
+    private StackDtoService stackDtoService;
+
+    @Inject
+    private FlowLogService flowLogService;
+
+    @Inject
+    private RegionAwareInternalCrnGeneratorFactory regionAwareInternalCrnGeneratorFactory;
+
+    @Inject
+    private RotateSaltPasswordService rotateSaltPasswordService;
+
+    public StackSaltStatusCheckerJob(Tracer tracer) {
+        super(tracer, "Stack Salt Status Checker Job");
+    }
+
+    @Override
+    protected Object getMdcContextObject() {
+        return stackDtoService.getStackViewById(getStackId());
+    }
+
+    @Override
+    protected void executeTracedJob(JobExecutionContext context) throws JobExecutionException {
+        if (flowLogService.isOtherFlowRunning(getStackId())) {
+            LOGGER.debug("StackSaltStatusCheckerJob cannot run, because flow is running for stack: {}", getStackId());
+            return;
+        }
+        try {
+            measure(() -> {
+                StackDto stack = stackDtoService.getById(getStackId());
+                rotateSaltPasswordService.validateRotateSaltPassword(stack);
+                Status stackStatus = stack.getStatus();
+                if (Status.getUnschedulableStatuses().contains(stackStatus)) {
+                    LOGGER.debug("Stack salt sync will be unscheduled, stack state is {}", stackStatus);
+                    jobService.unschedule(context.getJobDetail().getKey());
+                } else if (null == stackStatus || ignoredStates().contains(stackStatus)) {
+                    LOGGER.debug("Stack salt sync is skipped, stack state is {}", stackStatus);
+                } else if (syncableStates().contains(stackStatus)) {
+                    RegionAwareInternalCrnGenerator dataHub = regionAwareInternalCrnGeneratorFactory.datahub();
+                    ThreadBasedUserCrnProvider.doAs(dataHub.getInternalCrnForServiceAsString(), () -> doSync(stack));
+                } else {
+                    LOGGER.warn("Unhandled stack status, {}", stackStatus);
+                }
+            }, LOGGER, "Check salt status took {} ms for stack {}.", getStackId());
+        } catch (BadRequestException e) {
+            LOGGER.info("StackSaltStatusCheckerJob cannot run, because validation failed for stack {} with message: {}", getStackId(), e.getMessage());
+            jobService.unschedule(context.getJobDetail().getKey());
```

Comment:
- if we unschedule because the account doesn't has the entitlement, what would happen if the account gets the entitlement? when would it reschedule?
- similarly, would it be rescheduled after an upfrade if the SB gets a newer version?

Reply:
It will be rescheduled in both cases when CB service is restarted